dashboard/app/controllers/courses_controller.rb

```diff
@@ -173,6 +173,11 @@ def check_plc_enrollment
   end
 
   def render_no_access
+    unless @unit_group.can_be_instructor?(current_user) || @unit_group.can_be_participant?(current_user)
+      authenticate_user!
```

Comment:
I'm not familiar with `authenticate_user` but why would we want to authenticate the user if we're going to return :no_access anyway?

Reply:
`authenticate_user` makes sure to redirect a signed out user to the sign in page but I have refactored to deal with that.